- Exercici 1

Crea una funció que donat un Array d’una dimensió, et faci un resum estadístic bàsic de les dades. Si detecta que l’array té més d’una dimensió, ha de mostrar un missatge d’error.







Crea una funció que donada una taula de dues dimensions (NxM), et calculi els totals per fila i els totals per columna.




In [8]:
import numpy as np

arr = np.array([1, 2, 3, 4, 5])

print(arr)


[1 2 3 4 5]


In [23]:
import numpy as np
arr = np.array([1, 2, 3, 4, 5])
resumen_estadistico(arr)

def resumen_estadistico(arr):
    if arr.ndim != 1:
        print("Error: L'array ha de tenir una dimensión")
        return

In [24]:
print("Resum estadistic del array:")
print("Media:", np.mean(arr))
print("Mediana:", np.median(arr))
print("Desviació estandard:", np.std(arr))
print("Minim:", np.min(arr))
print("Maxim:", np.max(arr))

Resum estadistic del array:
Media: 3.0
Mediana: 3.0
Desviació estandard: 1.4142135623730951
Minim: 1
Maxim: 5


In [25]:
arr = np.array([[1, 2], [3, 4]])
resumen_estadistico(arr)

Error: L'array ha de tenir una dimensión


- Exercici 2

Crea una funció que et generi un quadrat NxN de nombres aleatoris entre el 0 i el 100.

In [27]:
import numpy as np

def generate_random_square_matrix(N):
    return np.random.randint(0, 101, size=(N, N))

In [28]:
generate_random_square_matrix(5)

array([[41, 12, 30, 51, 45],
       [ 3, 98, 98, 67, 85],
       [46, 17, 25, 39, 90],
       [66, 40, 61, 99, 50],
       [80, 46, 93, 88, 98]])

- Exercici 3
Crea una funció que donada una taula de dues dimensions (NxM), et calculi els totals per fila i els totals per columna.

In [29]:
import numpy as np

def calcula_totals(matriu):
    """
    Calcula els totals per fila i per columna d'una matriu.
    
    :param matriu: una taula de dues dimensions (NxM)
    :return: una tupla amb dos arrays de NumPy amb els totals per fila i per columna
    """
    totals_fila = np.sum(matriu, axis=1)
    totals_columna = np.sum(matriu, axis=0)
    return (totals_fila, totals_columna)

In [30]:
matriu = np.array([[8, 9, 7],
                   [6, 5, 6],
                   [3, 7, 1]])

In [31]:
totals_fila, totals_columna = calcula_totals(matriu)

print("Totals per fila:", totals_fila)
print("Totals per columna:", totals_columna)

Totals per fila: [24 17 11]
Totals per columna: [17 21 14]


- Exercici 4

Implementa manualment una funció que calculi el coeficient de correlació. Informa-te'n sobre els seus usos i interpretació.

In [32]:
import numpy as np

def coeficient_correlacio(x, y):
    """
    Calcula el coeficient de correlació entre dues variables x i y.
    
    :param x: un array NumPy amb les dades de la variable x
    :param y: un array NumPy amb les dades de la variable y
    :return: el coeficient de correlació entre x i y
    """
    mitjana_x = np.mean(x)
    mitjana_y = np.mean(y)
    desv_est_x = np.std(x)
    desv_est_y = np.std(y)
    cov = np.sum((x - mitjana_x) * (y - mitjana_y))
    return cov / (len(x) * desv_est_x * desv_est_y)

In [33]:
x = np.array([5,4,3,2,1,6])
y = np.array([3,2,3,2,3,2,])

In [34]:
coef = coeficient_correlacio(x, y)

print("Coeficient de correlació:", coef)

Coeficient de correlació: -0.29277002188455997


NO HI HA RELACIÖ ENTRE ELLES PERQUÈ  EL VALOR ES -0.29277002188455997